lib/src/shortest_path.dart

```diff
@@ -111,12 +110,13 @@ Map<T, List<T>> _shortestPaths<T>(
       assert(existingPath == null ||
           existingPath.length <= (currentPathLength + 1));
 
-      if (existingPath == null) {
-        final newOption = List<T>(currentPathLength + 1)
-          ..setRange(0, currentPathLength, currentPath)
-          ..[currentPathLength] = edge;
+      if (existingPath == null && currentPath != null) {
+        final newOption = [
+          ...currentPath,
+          edge,
+        ];
```

Comment:
```suggestion
      if (existingPath == null) {
        final newOption = [
          ...?currentPath,
          edge,
        ];
```

Reply:
This could also be implemented with a `List.generate` to make it be a fixed length list, which might be better